app/scripts/modules/kubernetes/src/v2/manifest/ManifestQos.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { get } from 'lodash';
+
+import { HoverablePopover } from '@spinnaker/core';
+import './manifestLabels.less';
+
+// TODO(dpeach) https://github.com/spinnaker/spinnaker/issues/3239
+export interface IManifestQosProps {
+  manifest?: {
+    status?: {
+      qosClass?: string
+    }
+  };
+}
+
+export class ManifestQos extends React.Component<IManifestQosProps> {
+  private qosClass(): string {
+    return get(this.props, ['manifest', 'status', 'qosClass'], 'Unknown');
+  }
+
+  private qosStyle(): string {
+    const qosClass = this.qosClass();
+    if (qosClass === 'Guaranteed') {
+      return 'success';
+    } else if (qosClass === 'BestEffort') {
+      return 'alert'
+    } else {
+      return 'warn'
+    }
+  }
+
+  private description = () => {
+    return (
+      <div>
+        <p>There are three QOS (Quality of Service) classes:</p>
+        <p><b>Guaranteed</b>: Pods are considered highest-priority, and will only be killed if they exceed their resource requests when other containers require their resources.</p>
+        <p><b>Burstable</b>: Pods have some minimum guaranteed resources and can exceed them, but are more likely to be killed than <b>Guaranteed</b> pods.</p>
+        <p><b>BestEffort</b>: These are the lowest-priority pods, and will be the first to be evicted when resources run out.</p>
+        <p>To understand how to set the QOS class of a pod, read <a href="https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/">the Kubernetes documentation</a>.</p>
```

Comment:
Is this something we could/should consider as something Spinnaker can "auto-add" to manifests/entire Spinnaker clusters? Or is our general policy to keep the amount of "manifest wrangling" we do to a minimum?

Reply:
If we build UIs for helping create common manifest YAMLs, I could expect a user could select the QOS class first, and then be prompted to input resource values. For example, if they select "Guaranteed", they only have to enter limits, not requests. If they select "BestEffort", there are no resources to input.

But, as it stands with regular manifests QOS classes are not a first class property, but rather a derived one, so there is little Spinnaker can do w/o changing input manifest format.